src/com/rath/rathbot/cmd/disc/actions/KickCmd.java

```diff
@@ -1,6 +1,104 @@
+
 package com.rath.rathbot.cmd.disc.actions;
 
+import com.rath.rathbot.RathBot;
+import com.rath.rathbot.cmd.PermissionsTable;
+import com.rath.rathbot.cmd.RBCommand;
+import com.rath.rathbot.msg.MessageHelper;
+
+import sx.blah.discord.api.IDiscordClient;
+import sx.blah.discord.handle.obj.IMessage;
+import sx.blah.discord.handle.obj.IUser;
+
+/**
+ * Kicks user by UID or @mention for a given reason.
+ * 
+ * @author Kami lehenbnw@gmail.com
+ * @author Tim Backus tbackus127@gmail.com
+ *
+ */
 
-public class KickCmd {
+public class KickCmd extends RBCommand {
+  
+  @Override
+  public String getCommandName() {
+    return "kick";
+  }
+  
+  @Override
+  public String getCommandDescription() {
+    return "(Mod only) Kicks a user defined by their UID or @mention for a given reason.";
+  }
+  
+  @Override
+  public String getCommandUsage() {
+    return "rb! kick <uid|@mention> <reason..>";
+  }
+  
+  @Override
+  public int permissionLevelRequired() {
+    return RBCommand.PERM_MODERATOR;
+  }
+  
+  @Override
+  public boolean requiresDirectMessage() {
+    return false;
+  }
+  
+  @Override
+  public boolean executeCommand(final IMessage msg, final String[] tokens, final int tokDepth) {
+    
+    // Ensures at least minimum valid arguments used.
+    if (tokens.length < 4) {
+      RathBot.sendMessage(msg.getChannel(), "Syntax Error! Usage: rb! kick <uid> <reason>");
+      return RBCommand.STOP_CMD_SEARCH;
+    }
+    
+    // Tests if the first argument of the command is an @mention or uid, then processes the argument accordingly.
+    long kickUserID = 0;
+    String userToken = tokens[tokDepth + 1];
+    if (userToken.matches("<@!?\\d+>")) {
```

Comment:
Actually, I feel like since we're using this block of code more than once, it should be in a separate class. Maybe create a new method `MessageHelper.getUIDFromToken`?

Reply:
I concur this pops up in A LOT of commands and it can be used by any command that takes a UID or @mention, even if the command *can't* take an @mention (like ReportCmd).